From b5b736e5e5e79fa21868512d0504be6c94dfc30a Mon Sep 17 00:00:00 2001
From: Jordan Hendricks <jordan.hendricks@joyent.com>
Date: Wed, 24 May 2017 22:03:50 +0000
Subject: [PATCH] joyent/node-manta#308 `mmpu commit` does not parse options
 joyent/node-manta#309 MPU tests are out of sync with Muskie master branch
 implementation Reviewed by: Approved by:

---
 bin/mmpu            | 3 +++
 lib/client.js       | 6 +++---
 test/client.test.js | 8 ++++----
 test/mmpu.test.js   | 8 ++++----
 4 files changed, 14 insertions(+), 11 deletions(-)

diff --git a/bin/mmpu b/bin/mmpu
index 640d8bb..accfba2 100755
--- a/bin/mmpu
+++ b/bin/mmpu
@@ -433,6 +433,9 @@ MMpu.prototype.do_commit = function do_commit(subcmd, opts, args, cb) {
 };
 
 
+MMpu.prototype.do_commit.options = manta.DEFAULT_CLI_OPTIONS;
+
+
 MMpu.prototype.do_commit.help = [
     'Commit a multipart upload.',
     '',
diff --git a/lib/client.js b/lib/client.js
index 5772c69..25c8dca 100644
--- a/lib/client.js
+++ b/lib/client.js
@@ -3361,7 +3361,7 @@ MantaClient.prototype.createUpload = function createUpload(p, opts, cb) {
 
         self.jsonClient.post(options, body, function (err2, req, res, obj) {
             if (err2) {
-                err2 = new Verror(err2, 'jsonClient POST');
+                err2 = new Verror(err2);
                 log.debug(err2);
                 cb(err2);
             } else {
@@ -3486,7 +3486,7 @@ MantaClient.prototype.abortUpload = function abortUpload(id, opts, cb) {
 
         self.jsonClient.post(options, function (err2, req, res, obj) {
             if (err2) {
-                err2 = new Verror(err2, 'jsonClient POST');
+                err2 = new Verror(err2);
                 log.debug(err2);
                 cb(err2);
             } else {
@@ -3593,7 +3593,7 @@ MantaClient.prototype.commitUpload = function commitUpload(id, p, opts, cb) {
         var body = { parts:  p };
         self.jsonClient.post(options, body, function (err2, req, res) {
             if (err2) {
-                err2 = new Verror(err2, 'jsonClient POST');
+                err2 = new Verror(err2);
                 log.debug(err2);
                 cb(err2);
             } else {
diff --git a/test/client.test.js b/test/client.test.js
index 3136dd1..29fd7d6 100644
--- a/test/client.test.js
+++ b/test/client.test.js
@@ -791,8 +791,8 @@ test('commit upload', function (t) {
             t.ifError(err2);
             t.ok(upload);
             t.equal(upload.id, UPLOAD1);
-            t.equal(upload.state, 'finalizing');
-            t.equal(upload.type, 'commit');
+            t.equal(upload.state, 'done');
+            t.equal(upload.result, 'committed');
 
             self.client.get(PATH1, function (err3, stream) {
                 t.ifError(err3);
@@ -834,8 +834,8 @@ test('abort upload', function (t) {
                 t.ifError(err3);
                 t.ok(upload);
                 t.equal(upload.id, UPLOAD2);
-                t.equal(upload.state, 'finalizing');
-                t.equal(upload.type, 'abort');
+                t.equal(upload.state, 'done');
+                t.equal(upload.result, 'aborted');
 
                 t.done();
             });
diff --git a/test/mmpu.test.js b/test/mmpu.test.js
index 2a69312..c5be2e5 100644
--- a/test/mmpu.test.js
+++ b/test/mmpu.test.js
@@ -508,8 +508,8 @@ test('mmpu commit C_ID C_ETAG0', function (t) {
                 cb(err);
             } else {
                 var upload = JSON.parse(info.stdout);
-                t.ok(upload.state, 'finalizing');
-                t.ok(upload.type, 'commit');
+                t.ok(upload.state, 'done');
+                t.ok(upload.result, 'committed');
                 cb();
             }
         });
@@ -585,8 +585,8 @@ test('mmpu abort A_ID', function (t) {
                 cb(err);
             } else {
                 var upload = JSON.parse(info.stdout);
-                t.ok(upload.state, 'finalizing');
-                t.ok(upload.type, 'abort');
+                t.ok(upload.state, 'done');
+                t.ok(upload.result, 'aborted');
                 cb();
             }
         });
-- 
2.21.0

